src/dmd/escape.d

```diff
@@ -789,26 +789,25 @@ ByRef:
         if (!(va && va.isScope()))
             notMaybeScope(v);
 
-        if ((v.storage_class & (STC.ref_ | STC.out_)) == 0 && p == sc.func)
+        if (!((v.storage_class & (STC.ref_ | STC.out_)) == 0 && p == sc.func))
```

Comment:
While a valid refactor, double negatives can be expressed much more intelligibly:

```suggestion
        if ((v.storage_class & (STC.ref_ | STC.out_)) || p != sc.func)
```